src/v/redpanda/application.cc

```diff
@@ -159,8 +159,57 @@ static void log_system_resources(
     }
 }
 
+namespace {
+
+static constexpr std::string_view community_msg = R"banner(
+
+Welcome to the Redpanda community!
+
+Documentation: https://docs.redpanda.com - Product documentation site
+GitHub Discussion: https://github.com/redpanda-data/redpanda/discussions - Longer, more involved discussions
+GitHub Issues: https://github.com/redpanda-data/redpanda/issues - Report and track issues with the codebase
+Support: https://support.redpanda.com - Contact the support team privately
+Product Feedback: https://redpanda.com/feedback - Let us know how we can improve your experience
+Slack: https://redpanda.com/slack - Chat about all things Redpanda. Join the conversation!
+Twitter: https://twitter.com/redpandadata - All the latest Redpanda news!
+
+)banner";
+
+} // anonymous namespace
+
 int application::run(int ac, char** av) {
-    init_env();
+    ss::app_template app(setup_app_config());
+    app.add_options()("version", po::bool_switch(), "print version and exit");
+    app.add_options()(
+      "redpanda-cfg",
+      po::value<std::string>(),
+      ".yaml file config for redpanda");
+
+    // Validate command line args using options registered by the app and
+    // seastar. Keep the resulting variables in a temporary map so they don't
+    // live for the lifetime of the application.
+    {
+        po::variables_map vm;
+        if (!cli_parser{
+              ac,
+              av,
+              cli_parser::app_opts{app.get_options_description()},
+              cli_parser::ss_opts{app.get_conf_file_options_description()},
+              _log}
+               .validate(vm)) {
+            return 1;
+        }
+        if (vm["version"].as<bool>()) {
+            std::cout << redpanda_version() << std::endl;
+            return 0;
+        }
+        if (vm.find("redpanda-cfg") == vm.end()) {
+            _log.error("Missing redpanda-cfg flag");
+            return 1;
+        }
+    }
```

Comment:
There are a few seastar specific help options/listing of loggers which will not work if we exit here. 

They are all processed in seastar's run/run_deprecated (see `app.get_conf_file_options_description()`). So this check probably needs to happen later.